aws-s3-transfer-manager/tests/download_objects_test.rs

```diff
@@ -403,3 +422,168 @@ async fn test_destination_dir_not_valid() {
     let err_str = format!("{}", DisplayErrorContext(err));
     assert!(err_str.contains("target is not a directory"));
 }
+
+#[tokio::test]
+async fn test_abort_on_handle_should_terminate_tasks_gracefully() {
+    let (_guard, rx) = capture_test_logs();
+
+    let bucket = MockBucket::builder()
+        .key_with_size("key1", 12)
+        .key_with_error("key2")
+        .key_with_size("key3", 7)
+        .build();
+
+    let client = mock_client!(aws_sdk_s3, RuleMode::MatchAny, bucket.rules().as_slice());
```

Comment:
Don't forget we need to configure an infallible HTTP client anywhere we are using mock client (and likely need to address this upstream but until then...)

Reply:
Yeah, good point. I created a thin wrapper macro in `test_common` and updated usages of `mock_client` throughout the codebase in d0159a3 and e78ecd1.